src/Datadog.Trace/AsyncLocalScopeManager.cs

```diff
@@ -21,17 +21,19 @@ public Scope Activate(Span span, bool finishOnClose = true)
         public void Close(Scope scope)
         {
             var current = _currentSpan.Get();
+
             if (current == null)
             {
-                _log.Error("Trying to close a null scope");
+                _log.Error("Trying to close a null scope.");
             }
-
-            if (current != scope)
+            else if (current != scope)
             {
-                _log.Warn("Current span doesn't match desactivated span");
+                _log.Warn("Specified scope is not the active scope.");
```

Comment:
So you can't close it?  If you're only allowed to close the current scope, why is a scope passed here at all (and why isn't this something like CloseCurrentScope()).  Not knowing the rest of the code base, I'd be curious what happens when something asks for this passed scope to be closed and it then isn't closed (i.e. does it just stick around forever?)

Reply:
Good point. I didn't think this one through. I do think the null check I added above is required, otherwise we could end up calling `_currentSpan.Set(current.Parent)` below and get a NRE. I'll revert this bit for now.

FWIW, I think this is just a sanity check, since users should call `Scope.Close()` from their code and not `AsyncLocalScopeManager.Close(Scope)`.